09_json/nickolee/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	puppy "github.com/anz-bank/go-course/09_json/nickolee/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/nickolee/pkg/puppy/store"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var out io.Writer = os.Stdout
+
+var (
+	args     = os.Args[1:]
+	fileName = kingpin.Flag("data", "This flag supplies a puppy JSON file name").Short('d').String()
+)
+
+func main() {
+	// parse flags and read json from file
+	fileData := parseFlagsAndLoadFile() // fileData var to store raw bytes read in from file
```

Comment:
I would separate flag parsing and loading a file
```go
func main() {
  kingpin.Parse()
  filedata := loadFile(*filename)
}
```

A larger application might have multiple files it needs to load, or other weird things it needs to do with the parsed flags. Separating out like this makes it
1. Easier to test each part of the main function
2. Gives an error more specific to exactly what went wrong.

Reply:
Have split this into 2 now. Good suggestion 🥇